extensions/resteasy-reactive/quarkus-resteasy-reactive/deployment/src/main/java/io/quarkus/resteasy/reactive/server/deployment/ResteasyReactiveProcessor.java

```diff
@@ -558,23 +554,27 @@ public Status isJava19OrHigher() {
             }
 
             serverEndpointIndexerBuilder.setMultipartReturnTypeIndexerExtension(new QuarkusMultipartReturnTypeHandler(
-                    generatedClassBuildItemBuildProducer, applicationClassPredicate, reflectiveClassBuildItemBuildProducer));
+                    generatedClassBuildItemBuildProducer, applicationClassPredicate, reflectiveClass));
             serverEndpointIndexerBuilder.setMultipartParameterIndexerExtension(new QuarkusMultipartParamHandler(
-                    generatedClassBuildItemBuildProducer, applicationClassPredicate, reflectiveClassBuildItemBuildProducer,
+                    generatedClassBuildItemBuildProducer, applicationClassPredicate, reflectiveClass,
                     bytecodeTransformerBuildItemBuildProducer));
             serverEndpointIndexer = serverEndpointIndexerBuilder.build();
 
             Map<String, List<EndpointConfig>> allMethods = new HashMap<>();
             for (ClassInfo i : scannedResources.values()) {
                 Optional<ResourceClass> endpoints = serverEndpointIndexer.createEndpoints(i, true);
                 if (endpoints.isPresent()) {
+                    ResourceClass resourceClass = endpoints.get();
                     if (singletonClasses.contains(i.name().toString())) {
-                        endpoints.get().setFactory(new SingletonBeanFactory<>(i.name().toString()));
+                        resourceClass.setFactory(new SingletonBeanFactory<>(i.name().toString()));
                     }
-                    resourceClasses.add(endpoints.get());
-                    for (ResourceMethod rm : endpoints.get().getMethods()) {
-                        addResourceMethodByPath(allMethods, endpoints.get().getPath(), i, rm);
+                    resourceClasses.add(resourceClass);
+                    for (ResourceMethod rm : resourceClass.getMethods()) {
+                        addResourceMethodByPath(allMethods, resourceClass.getPath(), i, rm);
                     }
+
+                    // Ensure native support of resource endpoints:
+                    reflectiveClass.produce(new ReflectiveClassBuildItem(false, true, false, resourceClass.getClassName()));
```

Comment:
@geoand this is the fix I added to solve the linked issue. However, as it works when the resource is not using a File class, I doubt this is the right fix... tho I could not find the right path.

Reply:
Indeed, this is not the fix we want to apply. I will  have a closer look later

Reply:
I converted this pull request in draft, just to reuse the tests and the clean up.